test/test_dictionary.cpp

```diff
@@ -478,3 +478,100 @@ TEST(Dictionary_UseAfterFree)
     }
     CHECK_EQUAL(q.count(), 1);
 }
+
+TEST(Dictionary_HashCollision)
+{
+    auto mask = Dictionary::set_hash_mask(0xFF);
+    Group g;
+    auto foos = g.add_table("Foo");
+    ColKey col_dict = foos->add_column_dictionary(type_Int, "dict");
+
+    auto foo = foos->create_object();
+    auto dict = foo.get_dictionary(col_dict);
+    for (int64_t i = 0; i < 100; i++) {
+        std::string key = "key" + util::to_string(i);
+        dict.insert(Mixed(key), i);
+    }
```

Comment:
Assuming I got the math right, assuming an even distribution from the hash function this is expected to have five hash collisions, and is very unlikely to have a double collision. Bumping up to 1000 entries would be an easy way to better exercise the code being tested (and indeed the test fails when I make that change).

Reply:
Well - the test gets 18 collisions of which 3 are double collisions. The error that @jbreams exposed prevented the test to handle many collisions.